src/NuGet.Core/NuGet.Protocol.Core.v3/Utility/GetDownloadResultUtility.cs

```diff
@@ -21,14 +24,19 @@ public static class GetDownloadResultUtility
            PackageIdentity identity,
            Uri uri,
            ISettings settings,
+           SourceCacheContext cacheContext,
            ILogger logger,
            CancellationToken token)
         {
+            // XXXXXXXXXXX This following assumption is false we constructed a Uri but we haven't attempted to use it yet
```

Comment:
I will remove the comment. It was related to the fact that -NoCache option is ignored, but I chose not to attempt to solve that issue in that PR. This PR is just for -DirectDownload.